### PR TITLE
RegEx issue with malformed HTML in addLazyloadClass

### PR DESCRIPTION
**Pull request by crystinutzaa**
Rocket issue https://github.com/wp-media/wp-rocket/issues/2706

**Reproduce the issue** ✅ 
Identified the issue on my localhost.

**Identify the root cause** ✅ 
@Tabrisrp made a good investigation and that regex seems to be off with malformed HTML.

**Scope a solution** ✅ 
Solution proposed by Remy is valid and seems much more suitable than the current one

**Effort** ✅ 
Effort `[XS]`

### CODE REVIEW

File: src/Image.php [hunk 1]
```diff
@@ -88,12 +88,14 @@ public function lazyloadBackgroundImages( $html, $buffer ) {
...
-		if ( preg_match( '#class=["\']?(?<classes>[^"\'>]*)["\']?#is', $element, $class ) ) {
+		if ( preg_match( '#class\s*=\s*(?<classes>["\'].*?["\']|[^\s]+)#is', $element, $class ) ) {
+			$contains_quotes  = preg_match( '#^(\'|").+(\'|")$#', $class['classes'] );
+			$class['classes'] = str_replace( [ '"', '\'' ], '', $class['classes'] );
 			if ( empty( $class['classes'] ) ) {
 				return str_replace( $class[0], 'class="rocket-lazyload"', $element );
 			}
 
-			$classes = str_replace( $class['classes'], $class['classes'] . ' rocket-lazyload', $class[0] );
+			$classes = str_replace( $class['classes'], ( ! $contains_quotes ? '"' : '' ) . $class['classes'] . ' rocket-lazyload' . ( ! $contains_quotes ? '"' : '' ), $class[0] );
```
Comment (hellofromtonya) [inline src/Image.php:98]: For this one, the default of the ternaries concatenates an empty `''` string. That's unnecessary and adds processing time. 

Instead, create a variable with the replacement class:

```suggestion
			$contains_quotes  = preg_match( '#^(\'|").+(\'|")$#', $class['classes'] );
			$replacement_class = $class['classes'] . ' rocket-lazyload';
			// Does not contain quotes.
			if ( ! preg_match( '#^(\'|").+(\'|")$#', $class['classes'] ) ) {
				$replacement_class = '"' . $replacement_class . '"';
			}
			$classes = str_replace( $class['classes'], $replacement_class, $class[0] );
```

**Comment by crystinutzaa** [comment]
Comments do not work for me in github.

**Comment by hellofromtonya** [comment]
@crystinutzaa Take a look at the refactoring I did for this PR. 

- Fixes malformed class attribute. For example:

```html
class = " class1    class2      "
```

is normalized to be:

```html
class="class1 class2"
```

- Grabs the original quotation character and then reapplies that when normalizing the class attribute values.

- Abstracts the tasks into private methods to make it more readable.

What do you think?

**Comment by crystinutzaa** [comment]
Thank you @hellofromtonya . Indeed it looks way better and cleaner.

**Comment by crystinutzaa** [comment]
@arunbasillal & @Davidacu this PR is ready for QA

**Pull request merged**